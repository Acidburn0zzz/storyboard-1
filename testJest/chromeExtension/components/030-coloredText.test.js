/* eslint-env jest */
/* eslint-disable no-extend-native */
import React from 'react';
import renderer from 'react-test-renderer';
import chalk from 'chalk';
import ColoredText from '../../../lib/chromeExtension/components/030-coloredText';

describe('ColoredText', () => {
  it('renders correctly with unstyled text', () => {
    const tree = renderer.create(
      <ColoredText text="Hello" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with styled text', () => {
    const text = `That's ${chalk.red('red')} and that's ${chalk.cyan.bold('very important')}`;
    const tree = renderer.create(
      <ColoredText text={text} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders nested styles correctly', () => {
    const text = `That's ${chalk.red(`${chalk.bold('very')} important`)}`;
    const tree = renderer.create(
      <ColoredText text={text} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('respects the style prop', () => {
    const tree = renderer.create(
      <ColoredText text="Hello" style={{ border: '1px solid black' }} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('respects the onClick prop', () => {
    const tree = renderer.create(
      <ColoredText text="Hello" onClick={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
