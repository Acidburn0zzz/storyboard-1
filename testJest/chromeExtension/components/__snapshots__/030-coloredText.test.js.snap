exports[`ColoredText renders correctly with styled text 1`] = `
<span
  onClick={undefined}
  style={undefined}>
  <span
    style={Object {}}>
    That\'s 
  </span>
  <span
    style={
      Object {
        "color": "#cc0000",
      }
    }>
    red
  </span>
  <span
    style={Object {}}>
     and that\'s 
  </span>
  <span
    style={
      Object {
        "color": "darkturquoise",
        "fontWeight": "bold",
      }
    }>
    very important
  </span>
</span>
------------HTML PREVIEW---------------
  <span
    style=>
    <span
      style="">
      That\'s 
    </span>
    <span
      style="color:#cc0000;">
      red
    </span>
    <span
      style="">
       and that\'s 
    </span>
    <span
      style="color:darkturquoise;font-weight:bold;">
      very important
    </span>
  </span>
`;

exports[`ColoredText renders correctly with unstyled text 1`] = `
<span
  onClick={undefined}
  style={Object {}}>
  Hello
</span>
------------HTML PREVIEW---------------
  <span
    style="">
    Hello
  </span>
`;

exports[`ColoredText renders nested styles correctly 1`] = `
<span
  onClick={undefined}
  style={undefined}>
  <span
    style={Object {}}>
    That\'s 
  </span>
  <span
    style={
      Object {
        "color": "#cc0000",
        "fontWeight": "bold",
      }
    }>
    very
  </span>
  <span
    style={
      Object {
        "color": "#cc0000",
      }
    }>
     important
  </span>
</span>
------------HTML PREVIEW---------------
  <span
    style=>
    <span
      style="">
      That\'s 
    </span>
    <span
      style="color:#cc0000;font-weight:bold;">
      very
    </span>
    <span
      style="color:#cc0000;">
       important
    </span>
  </span>
`;

exports[`ColoredText respects the onClick prop 1`] = `
<span
  onClick={[Function]}
  style={Object {}}>
  Hello
</span>
------------HTML PREVIEW---------------
  <span
    style="">
    Hello
  </span>
`;

exports[`ColoredText respects the style prop 1`] = `
<span
  onClick={undefined}
  style={
    Object {
      "border": "1px solid black",
    }
  }>
  Hello
</span>
------------HTML PREVIEW---------------
  <span
    style="border:1px solid black;">
    Hello
  </span>
`;
